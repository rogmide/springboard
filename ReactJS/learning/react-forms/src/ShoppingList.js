import React, { useState } from "react";
import Item from "./Item";
import NewItemForm from "./NewItemForm";

const ShoppingList = () => {
  const INITIAL_STATE = [
    { id: 1, name: "Peanut Butter", qty: 2 },
    { id: 2, name: "Whole Milk", qty: 1 },
  ];

  const [item, setItem] = useState(INITIAL_STATE);

  const addItem = (name, qty) => {
    setItem((item) => [...item, { name, qty }]);
  };

  return (
    <div>
      <h3>Shopping List</h3>
      <NewItemForm addItem={addItem} />
      {item.map(({ id, name, qty }) => (
        <Item id={id} name={name} qty={qty} key={id} />
      ))}
    </div>
  );
};

export default ShoppingList;
