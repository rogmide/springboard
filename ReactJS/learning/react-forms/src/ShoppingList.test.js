import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ShoppingList from "./ShoppingList";

it("render without crashing", function () {
  render(<ShoppingList />);
});

it("should match snapshot", () => {
  const { asFragment } = render(<ShoppingList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a new item", () => {
  const { queryByText, getByLabelText } = render(<ShoppingList />);
  const input = getByLabelText("Product:");
  const btn = queryByText("Add Items");
  expect(queryByText("Product Name: Chocolate Milk")).not.toBeInTheDocument();
  // mocking a input on the app passing a obj
  fireEvent.change(input, { target: { value: "Chocolate Milk" } });
  fireEvent.click(btn);
  expect(queryByText("Product Name: Chocolate Milk")).toBeInTheDocument();
});
