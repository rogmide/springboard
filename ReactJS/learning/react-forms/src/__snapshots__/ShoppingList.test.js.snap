// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Shopping List
    </h3>
    <form>
      <label
        for="name"
      >
        Product: 
      </label>
      <input
        id="name"
        name="name"
        placeholder="Item Name"
        type="text"
        value=""
      />
      <label
        for="qty"
      >
        Quantity: 
      </label>
      <input
        id="qty"
        max="10"
        min="1"
        name="qty"
        type="range"
        value=""
      />
      <button>
        Add Items
      </button>
    </form>
    <ul>
      <li>
        Product Name: Peanut Butter
      </li>
      <li>
        Quantity: 2
      </li>
    </ul>
    <ul>
      <li>
        Product Name: Whole Milk
      </li>
      <li>
        Quantity: 1
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
